arch/arm64/boot/dts/qcom/msm8953-huawei-milan.dts

```diff
@@ -0,0 +1,549 @@
+#include <dt-bindings/leds/common.h>
+
+
+/ {
+	model = "Huawei Maimang 5 / Nova (Plus) / G9 (Plus)";
+	compatible = "huawei,milan", "qcom,msm8953", "lk2nd,device";
+	qcom,msm-id = <293 0x0>;
+	qcom,board-id = <8000 0>, <8001 0>, <8002 0>, <8003 0>, <8016 0>,
+			<8017 0>, <8018 0>, <8019 0>, <8032 0>, <8033 0>,
+			<8035 0>, <8048 0>, <8049 0>, <8050 0>, <8051 0>,
+			<8066 0>, <8064 0>, <8067 0>, <8080 0>, <8082 0>,
+			<8083 0>, <8098 0>, <8099 0>, <8114 0>, <8130 0>;
+	chassis-type = "handset";
+	
+	battery: battery {
+		compatible = "simple-battery";
+		charge-full-design-microamp-hours = <3270000>;
+		constant-charge-current-max-microamp = <2000000>;
+		voltage-min-design-microvolt = <3400000>;
+		voltage-max-design-microvolt = <4400000>;
+	};
+	
+	vph_pwr: vph-pwr-regulator {
+		compatible = "regulator-fixed";
+		regulator-name = "vph_pwr";
+		regulator-always-on;
+		regulator-boot-on;
+	};
+
+	chosen {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		framebuffer@90001000 {
+			compatible = "simple-framebuffer";
+			reg = <0x00 0x90001000 0x00 0x5eec00>;
+			width = <0x438>;
+			height = <0x780>;
+			stride = <0xca8>;
+			format = "r8g8b8";
+
+			power-domains = <&gcc MDSS_GDSC>;
+
+			clocks = <&gcc GCC_MDSS_AHB_CLK>,
+					 <&gcc GCC_MDSS_AXI_CLK>,
+					 <&gcc GCC_MDSS_VSYNC_CLK>,
+					 <&gcc GCC_MDSS_MDP_CLK>,
+					 <&gcc GCC_MDSS_BYTE0_CLK>,
+					 <&gcc GCC_MDSS_PCLK0_CLK>,
+					 <&gcc GCC_MDSS_ESC0_CLK>;
+		};
+	};
+
+	gpio-keys {
+		compatible = "gpio-keys";
+
+		pinctrl-0 = <&gpio_key_default>;
+		pinctrl-names = "default";
+
+		key-volume-up {
+			label = "Volume Up";
+			gpios = <&tlmm 85 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_VOLUMEUP>;
+		};
+	};
+
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		other_ext_mem: other_ext_region@83100000 {
+			no-map;
+			reg = <0x0 0x83100000 0x0 0x3700000>;
+		};
+
+		/delete-node/qseecom@85b00000;
+		qseecom_mem: qseecom@f5800000 {
+			reg = <0x0 0xf5800000 0x0 0x1000000>;
+			no-map;
+		};
+
+		secure_mem: secure_region@f6800000 {
+			compatible = "shared-dma-pool";
+			reusable;
+			alignment = <0 0x400000>;
+			size = <0 0x09800000>;
+		};
+
+		adsp_mem: adsp_region@f5400000 {
+			compatible = "shared-dma-pool";
+			reusable;
+			size = <0 0x400000>;
+		};
+
+		gpu_mem: gpu_region@8f000000 {
+			compatible = "shared-dma-pool";
+			reusable;
+			alloc-ranges = <0x0 0x80000000 0x0 0x10000000>;
+			alignment = <0 0x400000>;
+			size = <0 0x800000>;
+		};
+
+		hwlog_tz_mem: hwlog_tz_region@92200000 {
+			linux,reserve-contiguous-region;
+			linux,reserve-region;
+			linux,remove-completely;
+			no-map;
+			reg = <0x0 0x92200000 0x0 0x0100000>;
+			label = "hwlog_tz_mem";
+		};
+
+		hwboot_fail_mem: hwboot_fail_region@86800000 {
+			linux,reserve-contiguous-region;
+			linux,reserve-region;
+			linux,remove-completely;
+			no-map;
+			reg = <0x0 0x86800000 0x0 0x0100000>;
+			label = "hwboot_fail_mem";
+		};
```

Comment:
Remove them, they are already defined in msm8953.dtsi

Reply:
but it is not same as in msm8953.dtsi defined

Reply:
because they are defined from 4.9. If you check difference between 4.9 3.18 the reserved memory parts will be different. The vendor kernels based on 3.18, you only needs to define reserved memory sections what are difference between msm-3.18 and Huawei vendor kernel. you don't need to care about difference between 3.18 4.9

Reply:
fixed